src/sync/partial_sync.cpp

```diff
@@ -176,7 +141,33 @@ void initialize_schema(Group& group)
     }
 
     // Remove any subscriptions no longer relevant
-    cleanup_subscriptions(group, timestamp_now());
+    cleanup_subscriptions(group, system_clock::now());
+}
+
+void ensure_partial_sync_schema_initialized(Realm& realm)
+{
+    auto was_in_read = realm.is_in_read_transaction();
+    auto cleanup = util::make_scope_exit([&]() noexcept {
+        if (realm.is_in_transaction())
+            realm.cancel_transaction();
+        if (!was_in_read)
+            realm.invalidate();
+    });
+
+    auto& group = realm.read_group();
+    // Check if the result sets table already has the expected number of columns
+    auto table = ObjectStore::table_for_object_type(group, result_sets_type_name);
+    if (table && table->size() >= 10)
```

Comment:
Perhaps move the magic constant 10 into a variable with docs, it is pretty opaque what it means.